config/sources/families/rk35xx.conf

```diff
@@ -1,42 +1,42 @@
 source "${BASH_SOURCE%/*}/include/rockchip64_common.inc"
 
-BOOTSOURCE='https://github.com/radxa/u-boot.git'
-BOOTBRANCH='branch:stable-4.19-rock3'
-BOOTPATCHDIR="u-boot-rk35xx"
-OVERLAY_PREFIX='rk35xx'
-
-if [[ "${BOOT_SOC}" == rk3566 ]]; then
-	CPUMAX=1800000
-	DDR_BLOB='rk35/rk3566_ddr_1056MHz_v1.10.bin'
-	BL31_BLOB='rk35/rk3568_bl31_v1.29.elf'
-elif [[ "${BOOT_SOC}" == rk3568 ]]; then
-	CPUMAX=1992000
-	DDR_BLOB='rk35/rk3568_ddr_1056MHz_v1.10.bin'
-	BL31_BLOB='rk35/rk3568_bl31_v1.29.elf'
+if [[ $BOARD == station-p2 || $BOARD == station-m2 ]]; then
+	BOOTSOURCE='https://github.com/150balbes/u-boot-rk'
+	BOOTBRANCH='branch:rk356x'
+	BOOTPATCHDIR="u-boot-station-p2"
+else
+	BOOTSOURCE='https://github.com/radxa/u-boot.git'
+	BOOTBRANCH='branch:stable-4.19-rock3'
+	BOOTPATCHDIR="u-boot-rk35xx"
+	OVERLAY_PREFIX='rk35xx'
+	UBOOT_COMPILER="aarch64-linux-gnu-"
+	UBOOT_USE_GCC='< 8.0'
 fi
 
 case $BRANCH in
 
 	legacy)
-		UBOOT_COMPILER="aarch64-linux-gnu-"
-		UBOOT_USE_GCC='< 8.0'
-		BOOTDIR='u-boot-rockchip64'
-		KERNELSOURCE='https://github.com/radxa/kernel'
-		KERNELBRANCH='branch:stable-4.19-rock3'
 		KERNELDIR='linux-rockchip64'
-		KERNELPATCHDIR='rk35xx-legacy'
 
+		if [[ $BOARD == station-p2 || $BOARD == station-m2 ]]; then
+			KERNELSOURCE='https://github.com/150balbes/rockchip-kernel'
+			KERNELBRANCH='branch:rk35xx'
+			LINUXCONFIG='linux-station-p2-'$BRANCH
+			LINUXFAMILY=station-p2
+			KERNELPATCHDIR='linux-station-p2-'$BRANCH
+			EXTRAWIFI="no"
+		else
+			KERNELSOURCE='https://github.com/radxa/kernel'
+			KERNELBRANCH='branch:stable-4.19-rock3'
+			LINUXCONFIG='linux-rock-3a-'$BRANCH
```

Comment:
I don't see the point in wasting time on unifying and adapting the legacy core (this is a temporary core for Armbian and will become unnecessary in a few months), let the manufacturers do it. Therefore, different source versions and different configurations can be used for each model. We will use the main core, common in sources and patches for all RK models. Therefore, all the features of the legacy software settings may differ significantly and should be separated. For example, legacy sources for rock3a are not acceptable and do not work on P2\M2.

Reply:
You are probably right with the waste of time unifying the sources although few months is most probably too optimistic prediction ;p

Reply:
> few months is most probably too optimistic prediction ;p

356x has a lot in common with 33xx, so the process of adaptation to the main core is very fast. Thanks to the work on 33xx, the main core already has most of the necessary for 356x. :)

Reply:
We had similar situation with 3399 legacy kernels, where it was too much work to get on a common ground.

But we have at least try to unify things where possible (for our own long term good, not so much about newbies reading the code). Having common config might not sounds like a great achievement, but it's worth trying. 

my 2c

Reply:
@150balbes most of the rock 3a config is my attempt to bring a lot of kernel configs from the current rockchip64 family to rock-3a. I think unifying makes more sense and potentially work together to make all 3 boxes work good. 
Probably it will be less work for both of us.

Not sure if it is worth investing in using https://github.com/rockchip-linux/kernel/tree/develop-4.19 but if you feel string about it we can do it.

Reply:
> @150balbes most of the rock 3a config is my attempt to bring a lot of kernel configs from the current rockchip64 family to rock-3a. I think unifying makes more sense and potentially work together to make all 3 boxes work good.  Probably it will be less work for both of us.

OK, then I will make a new PR taking into account your opinion, after that you will check its work and if there are no comments, we will send it to the main branch.


> Not sure if it is worth investing in using https://github.com/rockchip-linux/kernel/tree/develop-4.19 but if you feel string about it we can do it.

The BSP core is significantly changed by each manufacturer for its specific model, so I don't think it's worth spending time explaining these options (4.19) into one general structure, it's enough that you can use your own version for each model and let the manufacturers refine their source versions themselves if they want these kernel versions to be used in Armbian for their equipment.